v2/clients/http/utils/common.go

```diff
@@ -81,29 +79,33 @@ func createRequest(ctx context.Context, httpMethod string, baseUrl string, reque
 	return req, nil
 }
 
-func createRequestWithRawData(ctx context.Context, httpMethod string, url string, data interface{}, encoding string) (*http.Request, errors.EdgeX) {
-	var err error
-	var encodedData []byte
+func createRequestWithRawData(ctx context.Context, httpMethod string, url string, data interface{}) (*http.Request, errors.EdgeX) {
+	jsonEncodedData, err := json.Marshal(data)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to encode input data to JSON", err)
+	}
 
-	if encoding == clients.ContentTypeJSON || encoding == "" {
-		encoding = clients.ContentTypeJSON
-		encodedData, err = json.Marshal(data)
-		if err != nil {
-			return nil, errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to encode input data to JSON", err)
-		}
-	} else if encoding == clients.ContentTypeCBOR {
-		encodedData, err = cbor.Marshal(data)
-		if err != nil {
-			return nil, errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to encode input data to CBOR", err)
-		}
+	content := FromContext(ctx, clients.ContentType)
+	if content == "" {
+		content = clients.ContentTypeJSON
 	}
 
+	req, err := http.NewRequest(httpMethod, url, bytes.NewReader(jsonEncodedData))
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.KindClientError, "failed to create a http request", err)
+	}
+	req.Header.Set(clients.ContentType, content)
+	req.Header.Set(clients.CorrelationHeader, correlatedId(ctx))
+	return req, nil
+}
+
+func createRequestWithEncodedData(ctx context.Context, httpMethod string, url string, data []byte, encoding string) (*http.Request, errors.EdgeX) {
 	content := FromContext(ctx, clients.ContentType)
 	if content == "" {
 		content = encoding
```

Comment:
we should use `encoding` instead of `FromContext(ctx, clients.ContentType)`, because the content type might be overridden by `AddEventRequest.Encode()`

```
contentType := encoding
if encoding == "" {
  contentType = FromContext(ctx, clients.ContentType)
}
```